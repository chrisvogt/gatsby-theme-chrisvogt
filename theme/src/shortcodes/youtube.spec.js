import React from 'react'
import renderer from 'react-test-renderer'
import YouTube from './youtube'

describe('YouTube Shortcode', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <YouTube
          title='Here, There And Everywhere (Piano Cover) by Chris Vogt'
          url='https://www.youtube-nocookie.com/embed/XJashBvI17A'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a default title if one is not provided', () => {
    const testRenderer = renderer.create(
      <YouTube url='https://www.youtube-nocookie.com/embed/XJashBvI17A' />
    )
    const testInstance = testRenderer.root
    expect(testInstance.findByType('iframe').props.title).toEqual('YouTube video')
  })
})
