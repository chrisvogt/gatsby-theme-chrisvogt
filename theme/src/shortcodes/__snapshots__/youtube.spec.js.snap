// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube Shortcode matches the snapshot 1`] = `
<div
  className="VideoWrapper css-0"
>
  <iframe
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="315"
    src="https://www.youtube-nocookie.com/embed/XJashBvI17A"
    title="Here, There And Everywhere (Piano Cover) by Chris Vogt"
    width="560"
  />
</div>
`;
